doc/metric_spec.md

```diff
@@ -1,112 +1,179 @@
-# Metrics specification:
+# Metrics specification
 
-This is a specification of metrics exposed by the ha_cluster exporter.
+This document describes the metrics exposed by `ha_cluster_exporter`.
 
-All metrics from the exporter start with the prefix `ha_cluster`
+General notes:
+- All the metrics are _namespaced_ with the prefix `ha_cluster`, which is followed by a _subsystem_, and both are in turn composed into a _Fully Qualified Name_ (FQN) of each metrics.
+- All the metrics and labels _names_ are in snake_case, as conventional with Prometheus. That said, as much as we'll try to keep this consistent throughout the project, the label _values_ may not actually follow this convention, though (e.g. value is a hostname).
+- All the metrics are timestamped with the Unix epoch time in milliseconds; in the provided examples, this value will always be `1234`.
+- Some metrics, like `ha_cluster_pacemaker_nodes`, `ha_cluster_pacemaker_resources`, share common traits:
+  - their labels contain the relevant data you may want to track or use for aggregation and filtering;
+  - either their value is `1`, or the line is absent altogether; this is because each line represents one entity of the cluster, but the exporter itself is stateless, i.e. we don't track the life-cycle of entities that do not exist anymore in the cluster.
 
-Below you have a complete specification, ordered by component.
 
-1. [pacemaker](#pacemaker)
-2. [drbd](#drbd)
-3. [sbd](#sbd)
-4. [corosyncl](#corosync)
+These are the currently implemented subsystems.
 
-# Pacemaker 
+1. [Pacemaker](#pacemaker)
+2. [Corosync](#corosync)
+3. [SBD](#sbd)
+4. [DRBD](#drbd)
 
-The Pacemaker cluster metrics are atomic metrics and represent and updated snapshot of the HA cluster, retrieved fetching the XML CIB of pacemaker.
 
-Some of the pacemaker metrics like `ha_cluster_node_resources` and `ha_cluster_nodes` metrics with labels share a common trait:
- 
-they can be either set to `1` or they are absent, this is because they track the real state of the cluster resources monitored.
+## Pacemaker 
 
-1. [ha_cluster_node_resources](#ha_cluster_node_resources)
-2. [ha_cluster_nodes](#ha_cluster_nodes)
-3. [ha_cluster_nodes_configured_total](#ha_cluster_nodes_configured_total)
-4. [ha_cluster_resources_configured_total](#ha_cluster_resources_configured_total)
+The Pacemaker subsystem collects an atomic snapshot of the HA cluster directly from the XML CIB of Pacemaker via `crm_mon`.
 
+0. [Sample](../test/pacemaker.metrics)
```

Comment:
perhaps here:

instead of put the 0. `sample`, I would perhaps move it at the end of the index with  a link `examples`

1. foo
2. bar
3. baf

examples: (link)